objects.js

```diff
@@ -3445,7 +3445,12 @@ SpriteMorph.prototype.setPosition = function (aPoint, justMe) {
 
 SpriteMorph.prototype.forward = function (steps) {
     var dest,
-        dist = steps * this.parent.scale || 0;
+        dist;
+    
+    if (isNaN(steps) || !isFinite(steps)) {
```

Comment:
I don't know how @jmoenig Feels about this, but I feel like this should be a shared function.

``` function defaultFiniteValue(num, default) {
     if (isNaN(steps) || !isFinite(steps)) {
        return default;
    } else {
        return value;
    }
}
```

Actually, you could probably write that with `?:`.

Then you can just do `steps = defaultValue(steps, 0)` which I think is much easier to read.

(Feel free to rename the function, but I do think it would clean things up.)